db/change_cache.go

```diff
@@ -941,7 +945,10 @@ func (h *LogPriorityQueue) Pop() interface{} {
 // ////// SKIPPED SEQUENCE QUEUE
 
 func (c *changeCache) RemoveSkipped(x uint64) error {
-	err := c.skippedSeqs.removeSeq(x)
+	numSkipped, err := c.skippedSeqs.removeSeq(x)
+	if numSkipped == 0 {
+		c.db.mutationListener.BroadcastSlowMode.CompareAndSwap(true, false)
+	}
 	return err
```

Comment:
This is probably worth an error check now? Although the only cases we really care about works with the code as-is whether there's an error check or not (empty list or not found sequence)

I'm thinking if we were to add an error case we'd want to leave the state of `BroadcastSlowMode` alone and not switch back to fast mode, given `return 0, err` is a common enough pattern I can see that being written instead of returning the current skipped seq length.

```suggestion
	if err != nil {
		return err
	}
	if numSkipped == 0 {
		c.db.mutationListener.BroadcastSlowMode.CompareAndSwap(true, false)
	}
	return nil
```

Reply:
Good point, will adjust this.